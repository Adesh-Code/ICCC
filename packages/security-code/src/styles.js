const styles = ({ tokens, primitives }) => ({
  "&.DhigSecurityCode": {
    width: "56px",
    height: "72px",
    borderTop: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`,
    borderBottom: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"][class*="MuiInputBase-inputAdornedEnd"]':
    {
      marginRight: "0"
    },
  "&.DhigSecurityCode[class*='MuiInput-root']:hover:before": {
    height: "2px",
    backgroundColor: tokens.colors["fill-engage"],
    boxShadow: "none",
    borderBottom: `${primitives.borderWidths.medium} solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode:hover": {
    boxShadow: "none"
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.primary.white}`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.primary.black}`
  },
  "&.DhigSecurityCode input:focus": {
    boxShadow: `0px 2px 0px 0px ${tokens.colors["ink-focus"]} inset, 0px 0px 0px 2px ${tokens.colors["ink-focus"]} inset`
  },
  '&.DhigSecurityCode[class*="MuiInputBase-root"]:not(.Mui-disabled)::before': {
    boxShadow: "none"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]:before': {
    width: "56px"
  },
  "&.DhigSecurityCode input": {
    flexGrow: "1"
  },
  '&.DhigSecurityCode [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "56px",
    height: "72px",
    fontSize: `${primitives.fontSizeFallbacks["6"]}`,
    fontWeight: primitives.fontWeights.default
  },
  '&.DhigSecurityCode--dense [class*="MuiInputBase-input"]': {
    padding: "0",
    width: "32px",
    height: "40px",
    fontSize: `${primitives.fontSizeFallbacks["3"]}`,
    fontWeight: primitives.fontWeights.default
  },
  "&.DhigSecurityCode--dense": {
    width: "32px",
    height: "40px"
  },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:before': {
    borderBottom: `1px solid ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode--error, &.DhigSecurityCode:first-of-type.DhigSecurityCode--error, &.DhigSecurityCode:last-child.DhigSecurityCode--error, &.DhigSecurityCode--error:before":
    {
      borderColor: primitives.colors.accent.clay
    },
  '&.DhigSecurityCode--error[class*="MuiInputBase-root"]:after': {
    boxShadow: `0px 1px 0px 0px ${primitives.colors.accent.clay}`
  },
  "&.DhigSecurityCode:first-of-type": {
    borderLeft: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child": {
    borderRight: `${primitives.borderWidths.medium} solid ${primitives.colors.tint["slate-450"]}`
  },
  "&.DhigSecurityCode:last-child [class*='MuiDivider-root'], &.DhigSecurityCode:focus-within [class*='MuiDivider-root']":
    {
      display: "none"
    }
});

export default styles;
