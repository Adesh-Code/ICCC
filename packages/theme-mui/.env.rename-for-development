# To load the MUI X Pro license for local development, rename this file to .env and replace with the actual license string
DHIG_MUIXPRO_ENTRY=PASTE_MUI_X_PRO_LICENSE_STRING_HERE