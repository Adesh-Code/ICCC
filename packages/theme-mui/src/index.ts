// Themes
export { default as themeDark } from "./themes/dark";
export { default as themeLight } from "./themes/light";

// Utilities
export { configureClassnameGenerator } from "./classname-generator";
export { setLicenseKey } from "./set-license-key";
