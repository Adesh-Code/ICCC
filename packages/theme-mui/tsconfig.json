{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "lib/esm",
    "resolveJsonModule": true,
    "declarationDir": "lib/types"
  },
  "include": ["src"]
}
