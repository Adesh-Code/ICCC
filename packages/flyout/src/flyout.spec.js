import React from "react";
import { fireEvent, render, screen } from "@testing-library/react";
import Flyout from "./flyout";
import MuiMenuItem from "@mui/material/MenuItem";
import Button from "@mui/material/Button";

import { themeLight } from "@iccc/theme-mui";
import { createTheme, ThemeProvider } from "@mui/material/styles";

const light = createTheme(themeLight);

const arrange = ({
  children = <Button>button</Button>,
  menuItems = [
    { children: "Item 1" },
    { children: "Item 2" },
    { children: "Item 3" }
  ],
  open = true,
  ...otherProps
} = {}) =>
  render(
    <ThemeProvider theme={light}>
      <Flyout {...otherProps} menuItems={menuItems} open={open}>
        {children}
      </Flyout>
    </ThemeProvider>
  );

describe("Flyout component", () => {
  it("renders", () => {
    arrange();
    const flyout = screen.getByRole("tooltip");
    expect(flyout).toMatchSnapshot();
  });

  it("accepts children", () => {
    const { getAllByRole } = arrange({
      children: (
        <>
          <MuiMenuItem>One</MuiMenuItem>
          <MuiMenuItem>Two</MuiMenuItem>
        </>
      )
    });
    const flyout = getAllByRole("menuitem")[0];
    fireEvent.mouseDown(flyout);
    expect(screen.getByText(`Two`)).toBeInTheDocument();
  });

  it("renders a selected icon", () => {
    arrange();
    const itemToSelect = screen.getAllByRole("menuitem")[0];
    fireEvent.click(itemToSelect);
    expect(itemToSelect).toHaveClass("Mui-selected");
  });
});
