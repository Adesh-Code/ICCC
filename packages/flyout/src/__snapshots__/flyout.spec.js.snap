// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flyout component renders 1`] = `
<div
  class="MuiTooltip-popper DhigFlyout MuiTooltip-popperInteractive emotion-0 MuiPopper-root"
  data-popper-escaped=""
  data-popper-placement="bottom"
  data-popper-reference-hidden=""
  id="mui-1"
  role="tooltip"
  style="position: absolute; top: 0px; left: 0px; margin: 0px; transform: translate(0px, 0px);"
>
  <div
    class="MuiTooltip-tooltip MuiTooltip-tooltipPlacementBottom emotion-1"
    style="opacity: 1; transform: scale(1, 1); transition: opacity 125ms cubic-bezier(0.4, 0, 0.2, 1) 0ms,transform 83ms cubic-bezier(0, 0, 0.4, 1) 0ms;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 emotion-2"
    >
      <ul
        class="MuiList-root MuiList-padding emotion-3"
        role="menu"
        tabindex="-1"
      >
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters Mui-focusVisible MuiMenuItem-root MuiMenuItem-gutters emotion-4"
          role="menuitem"
          tabindex="0"
        >
          Item 1
        </li>
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters emotion-4"
          role="menuitem"
          tabindex="0"
        >
          Item 2
        </li>
        <li
          class="MuiButtonBase-root MuiMenuItem-root MuiMenuItem-gutters MuiMenuItem-root MuiMenuItem-gutters emotion-4"
          role="menuitem"
          tabindex="0"
        >
          Item 3
        </li>
      </ul>
    </div>
  </div>
</div>
`;
